app/src/components/market/sections/market_buy/market_buy.tsx

```diff
@@ -164,8 +166,8 @@ const MarketBuyWrapper: React.FC<Props> = (props: Props) => {
   const sharesTotal = formatBigNumber(tradedShares, collateral.decimals)
   const total = `${sharesTotal} Shares`
 
-  const hasZeroAllowance = allowance && allowance.isZero()
-  const showSetAllowance = allowanceFinished || hasZeroAllowance || !hasEnoughAllowance
+  const showSetAllowance =
+    allowanceFinished || hasZeroAllowance === Ternary.True || hasEnoughAllowance === Ternary.False
```

Comment:
Set allowance is shown if:

- We are sure that the user has zero allowance
- We are sure that the user doesn't have enough allowance

Same comment aplies in the pooling page.